hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/DevSources.java

```diff
@@ -0,0 +1,187 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.pipeline;
+
+import com.hazelcast.jet.function.SupplierEx;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Easy-to-use sources intended to be used during pipeline development.
+ * It's often convenient to have easy-to-use sources with well defined
+ * behaviour when developing a streaming application.
+ *
+ */
+public final class DevSources {
+    private static final AtomicLong COUNTER = new AtomicLong();
+    private static final String NAME_PREFIX = "dev-source-";
+
+    private DevSources() {
+
+    }
+
+    /**
+     * Streaming source which emit with fixed rate. It attempts to compensate for various system hiccups so the rate
+     * over a period of time is constant.
+     *
+     * Each item is a local wall-clock timestamp.
+     * The source is not distributed it means only a single Jet instance will emit items.
+     *
+     * @param period period which to emit
+     * @param timeUnit units for period
+     * @return source emitting with a fixed rate
+     */
+    public static StreamSource<Long> fixedRate(long period, TimeUnit timeUnit) {
+        return fixedRate(period, timeUnit, System::currentTimeMillis);
+    }
+
+    /**
+     * Streaming source which emits with specified minimum delay between items.
+     *
+     * It won't emit unless elapsed time since last emit is at least the specified delay. There is no upper bound
+     * on the delay at that's partially driven by the Jet engine. In practice it will behave similar to fixedDelay.
+     *
+     * The source is not distributed it means only a single Jet instance will emit items.
+     *
+     * @param delay minimum delay between emitting
+     * @param timeUnit units of the delay
+     * @return source emitting with a specified minimum delay
+     */
+    public static StreamSource<Long> minimumDelay(long delay, TimeUnit timeUnit) {
+        return minimumDelay(delay, timeUnit, System::currentTimeMillis);
+    }
+
+    /**
+     * Streaming source which emit items at fixed rate. It attempts to compensate for various system hiccups so
+     * the rate over a period of time is constant.
+     *
+     * You have to provide your own item supplier.
+     * The source is not distributed it means only a single Jet instance will emit items.
+     *
+     * @param period period which to emit
+     * @param timeUnit units for period
+     * @param itemSupplier supplier of items to be emitted. It's called whenever a new item is about to be emitted
+     * @param <T> type of emitted items
+     * @return source emitting with a fixed rate
+     */
+    public static <T> StreamSource<T> fixedRate(long period, TimeUnit timeUnit, SupplierEx<T> itemSupplier) {
```

Comment:
All functional interfaces must specify variance, in a boilerplate fashion (PECS). So, `SupplierEx<? extends T>` in this case.